docs/DEV.md

```diff
@@ -0,0 +1,45 @@
+# Hush Line Dev
+
+| Contents |
+|-|
+| 1. [Local Development](#local-development) |
+
+## Local Development
+
+This guide preps your machine to run Hush Line locally using the included `Makefile`.
+
+### Mac
+
+| Tested Platform | OS Version | Browser | Status | Date | Notes |
+|-|-|-|-|-|-|
+| Macbook M2 | OSX 13.2.1 | Firefox 124.0.2 | ✅ | Apr. 2024 | |
+| Macbook M1 | OSX 14.4.1 | Firefox 124.0.2 | ✅ | Apr. 2024 | |
+| Macbook M1 | OSX 14.4.1 | Safari 17.4.1 | ☑️ | Apr. 2024 | App starts but a CSRF token mismatch blocks registration. |
+
+#### Install Packages
+- `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`
+- `eval "$(/opt/homebrew/bin/brew shellenv)"`
+- `brew install python git git-lfs redis rust poetry`
+
+#### Clone the Repo
+- `git clone https://github.com/scidsg/hushline.git`
+- `cd hushline`
+- `git switch makefile` # Temporary until we merge back into Main
+- `/opt/homebrew/bin/python3 -m venv venv`
+- `source venv/bin/activate`
+- `poetry install`
+- `source env.sh`
+- `poetry run flask db upgrade` _Optional_
```

Comment:
This line and the next should be commented with "(Production migration)" and "(Local initialization)" or something similar since they are mutually exclusive and it should be clear why one is picked over the other

Reply:
Going to resolve this for now, as I think we should have a sprint that specifically looks at our developer documentation. 

The instructions' current state is what a few of us have gotten to work when setting up the environment. Other deviations from this path have failed. So, given that, they can probably be refined, but they are working and sufficient for the stage we're in. Closing the ticket to continue progress on other efforts.